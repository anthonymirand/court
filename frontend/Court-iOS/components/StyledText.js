import React from 'react';
import { Text } from 'react-native';

export class RegularText extends React.Component {
  render() {
    return <Text {...this.props} style={[this.props.style, { fontFamily: 'orkney-regular' }]} />;
  }
}

export class MediumText extends React.Component {
  render() {
    return <Text {...this.props} style={[this.props.style, { fontFamily: 'orkney-medium' }]} />;
  }
}

export class BoldText extends React.Component {
  render() {
    return <Text {...this.props} style={[this.props.style, { fontFamily: 'orkney-bold' }]} />;
  }
}

export class LightText extends React.Component {
  render() {
    return <Text {...this.props} style={[this.props.style, { fontFamily: 'orkney-light' }]} />;
  }
}
