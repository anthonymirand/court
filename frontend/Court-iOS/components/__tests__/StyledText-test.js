import 'react-native';
import React from 'react';
import { RegularText } from '../StyledText';
import renderer from 'react-test-renderer';

it('Renders Styled text', () => {
  // TODO(rivmist): change this to our custom text once loaded
  const tree = renderer.create(<RegularText>Snapshot test!</RegularText>).toJSON();

  expect(tree).toMatchSnapshot();
});
