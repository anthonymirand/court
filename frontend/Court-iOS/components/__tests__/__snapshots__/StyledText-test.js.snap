// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Styled text 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "orkney-regular",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;
